examples/dynamic-bridge-app/linux/include/data-model/DataModel.h

```diff
@@ -6,12 +6,29 @@
 #include <app/data-model/Nullable.h>
 
 #include <iterator>
+#include <type_traits>
 #include <vector>
 
 namespace chip {
 namespace app {
 namespace DataModel {
 
+// Container requirements:
+// To encode and decode as a string the following must be present
+// typedef T* pointer;
+// void assign(pointer, pointer);
+// pointer data();
+// size_type size();
+// T must be byte sized
+template <typename X>
+class AsString : public X
```

Comment:
Rather than viewing this as a cast, can we view these as just another set of types used by consumers when interacting with Matter data model types? i.e `StringContainer` and `ListContainer`? 

Seeing `AsList` and `AsString` intermingled with other more straightforward types feels odd.